sphinx_reredirects/__init__.py

```diff
@@ -160,3 +163,22 @@ def _render_redirect_template(self, to_uri: str) -> str:
         content = Template(redirect_template).substitute({"to_uri": to_uri})
 
         return content
+
+
+def collect_redirects_for_linkcheck(app):
+    if not isinstance(app.builder, CheckExternalLinksBuilder):
+        return
+
+    rr = Reredirects(app)
+    redirects = rr.grab_redirects()
+
+    for docname, target in redirects.items():
+        if new_target := app.emit_firstresult("linkcheck-process-uri", target):
```

Comment:
In which use case this `if` will apply? Or what this `if` tries to solve? Thanks.

Reply:
See https://www.sphinx-doc.org/en/master/extdev/event_callbacks.html#event-linkcheck-process-uri

Apparently, this event was added in https://github.com/sphinx-doc/sphinx/issues/9016 to fix an issue with GitHub, though that was later removed in https://github.com/sphinx-doc/sphinx/issues/9435. It's not used in Sphinx itself right now, though it may be used by extensions. (This part of the function is just mimicking the native link checker's logic.)